android/java/org/chromium/chrome/browser/brave_stats/BraveStatsBottomSheetDialogFragment.java

```diff
@@ -177,13 +189,71 @@ public void onClick(View v) {
             }
         });
         updateBraveStatsLayoutAsync();
+        updateNotificationView(view);
 
         dialog.setContentView(view);
         ViewParent parent = view.getParent();
         ((View)parent).getLayoutParams().height = ViewGroup.LayoutParams.MATCH_PARENT;
 
     }
 
+    public void onRequestPermissionsResult(
+            int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            statsNotificationView.setVisibility(View.GONE);
+        }
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (hasPermission(getActivity(), PermissionConstants.NOTIFICATION_PERMISSION)) {
+            statsNotificationView.setVisibility(View.GONE);
+        } else {
+            statsNotificationView.setVisibility(View.VISIBLE);
+        }
+    }
+
+    private void updateNotificationView(View view) {
+        ImageView btnDismiss = view.findViewById(R.id.button_dismiss);
+        btnDismiss.setOnClickListener(v -> { statsNotificationView.setVisibility(View.GONE); });
+        View notificationOnButton = view.findViewById(R.id.notification_on_button);
+        notificationOnButton.setOnClickListener(v -> {
+            if (getActivity().shouldShowRequestPermissionRationale(
+                        PermissionConstants.NOTIFICATION_PERMISSION)) {
+                // Last time don't allow selected in permission dialog, then enable through setting
+                redirectToSettingPage();
+            } else {
+                // 1st time request permission
+                ActivityCompat.requestPermissions(getActivity(),
+                        new String[] {PermissionConstants.NOTIFICATION_PERMISSION}, 1);
+            }
+        });
+    }
+
+    private void redirectToSettingPage() {
+        Intent intent = new Intent();
+        intent.setAction(Settings.ACTION_APP_NOTIFICATION_SETTINGS);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+        // for Android 5-7
+        intent.putExtra("app_package", getActivity().getPackageName());
+        intent.putExtra("app_uid", getActivity().getApplicationInfo().uid);
```

Comment:
nit: forgot to mention earlier, would be great if these two can also be replaced with constants, I wasn't able to find with a quick search. Feel free to resolve this comment yourself.

Reply:
updated